RELEASE.txt

```diff
@@ -18,64 +18,82 @@ make a PR to update these notes after you are done with the release. ;-)
 
   1. Review and cleanup ``doc/release/release_dev.txt``.
 
-  2. Make a list of merges and contributors by running
+  2. Make a list of merges, contributors, and reviewers by running
      ``doc/release/contribs.py <tag of previous release>``.
 
-  3. Paste this list at the end of the ``release_dev.txt``. Scan the PR titles
-     for highlights, deprecations, and API changes, and mention these in the
-     relevant sections of the notes.
+  3. Paste this list at the end of the ``release_dev.txt``.
 
-  4. Rename to ``doc/release/release_<major>.<minor>.txt``
+  4. Scan the PR titles for highlights, deprecations, API changes,
+     and bugfixes, and mention these in the relevant sections of the notes.
+     Try to present the information in an expressive way by mentioning
+     the affected functions, elaborating on the changes and their
+     consequences. If possible, organize semantically close PRs in groups.
 
-  5. Copy ``doc/release/release_template.txt`` to
+  5. Rename the file to ``doc/release/release_<major>.<minor>.txt``
+
+  6. Copy ``doc/release/release_template.txt`` to
      ``doc/release/release_dev.txt`` for the next release.
 
-  6. Copy relevant deprecations from ``release_<major>_<minor>.txt``
+  7. Copy relevant deprecations from ``release_<major>_<minor>.txt``
      to ``release_dev.txt``.
 
-- Update the version number in ``skimage/__init__.py`` and commit.
+- Submit the release notes for review by other project maintainers:
+
+  - Create a temporary PR from v<major>.<minor>.x branch to master
+    (at this point, the difference should show the full contents of the
+     release notes).
+  - Discuss, and make the changes directly to v<major>.<minor>.x branch.
+  - Once the consensus is found, close the PR and go to the next step.
+
+- Update the version number to stable in ``skimage/__init__.py`` and commit.
 
 - Update the docs:
 
   - Edit ``doc/source/_static/docversions.js`` and commit
   - Build a clean version of the docs.  Run ``python setup.py install`` in the
-    root dir, then ``rm -rf build; make html`` in the docs.
+    root dir, then ``rm -rf build; make html`` in the ``doc`` dir.
   - Build using ``make gh-pages``.
   - Update the symlink to ``stable``.
   - Push upstream: ``git push origin gh-pages`` in ``doc/gh-pages``.
 
 - Add the version number as a tag in git::
 
-   git tag -s [-u <key-id>] v<major>.<minor>.0
+   git tag -s -m <github_release_message> [-u <key-id>] v<major>.<minor>.0
```

Comment:
What should `<github_release_message>` be?

Reply:
Currently, we're just writing something like "Version 0.xx.y", but using the full release notes could be better (AFAIK, many teams (PyTorch, TensorFlow, etc) are doing this).

Reply:
You can take a look here, for example, - https://github.com/tensorflow/tensorflow/releases .

Reply:
Let's discuss what we'd like to see here. The full notes are, probably, too heavy. I'd suggest to cut the header (the one with a link to scikit-image.org) and the list of contributors.

Reply:
@soupault I agree with your points. You can add this here or in a future PR afaic. I don't want to hold up this PR for this.